drivers/platform/maxim/max32670/maxim_spi.c

```diff
@@ -199,12 +175,23 @@ int32_t max_spi_transfer(struct no_os_spi_desc *desc,
 			 struct no_os_spi_msg *msgs,
 			 uint32_t len)
 {
+	static uint32_t last_slave_id[MXC_SPI_INSTANCES];
 	mxc_spi_req_t req;
+	uint32_t slave_id;
 	int32_t ret;
 
 	if (!desc || !msgs)
 		return -EINVAL;
 
+	slave_id = desc->chip_select;
```

Comment:
slave_id should be a combination of device_id and chip_select

Reply:
like here: https://github.com/analogdevicesinc/no-OS/pull/1502/commits/3743b17170792b61507e2ffcfe5e3543b2c1a5d2#diff-9072e7b9877324e833f75a13a501eb0af2c8a032d8280e44e9f583bd3c8e5661R230

Reply:
`last_slave_id` is now an array, so the behavior is the same, since we save the chip_select number for each SPI interface individually. I think this is better than the current implementation, since there could be multiple SPI peripherals, and the following transfer sequence may occur:

[SPI0 | CS0] ... [SPI1 | CS0] ... [SPI0 | CS0]

In this case, SPI0 will be reconfigured before the last transfer, even though it's not needed, since we're talking with the same device as in the first transfer.

Reply:
true, you're right, i guess we need to change the other implementations to follow this logic, they are now too aggressive with changing the settings :D